src/sentry/tasks/commit_context.py

```diff
@@ -33,23 +34,32 @@
 logger = logging.getLogger(__name__)
 
 
-def queue_comment_task_if_needed(commit: Commit, group_owner: GroupOwner):
+def queue_comment_task_if_needed(
+    commit: Commit, group_owner: GroupOwner, repo: Repository, installation: IntegrationInstallation
+):
     from sentry.tasks.integrations.github.pr_comment import comment_workflow
 
     logger.info(
         "github.pr_comment.queue_comment_check",
         extra={"organization_id": commit.organization_id, "merge_commit_sha": commit.key},
     )
 
+    response = installation.get_client().get_pullrequest_from_commit(repo=repo.name, sha=commit.key)
+
+    if not (response.status_code == 200 and isinstance(response, list) and len(response) == 1):
+        # the response should return a single PR, return if multiple
+        return
```

Comment:
return or raise?

Reply:
Rather than the validation here, maybe we add the validation in the client?
Then we either return a single PR (as opposed to a list) or not

Reply:
either case, we should emit a log/stat so we can track those cases too

Reply:
hmm okay i'll address this in a followup